### PR TITLE
CHANGE expose vm's ui state as stateflow

### PR DESCRIPTION
**Pull request by Frank1234**
expose vm's ui state as stateflow:
- to not have to set a default twice
- this also solves a possible flickering default state bug, when navigating to this screen from backstack and a state change from the viewmodel is preceded by the default state for a split second.

### CODE REVIEW

File: core/navigation/src/main/kotlin/nl/q42/template/navigation/viewmodel/InitNavigator.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ import com.ramcosta.composedestinations.navigation.DestinationsNavigator
 @Composable
 fun InitNavigator(destinationsNavigator: DestinationsNavigator, routeNavigator: RouteNavigator) {
 
-    val viewState by routeNavigator.navigationState.collectAsStateWithLifecycle(initialValue = NavigationState.Idle)
+    val viewState by routeNavigator.navigationState.collectAsStateWithLifecycle()
```
Comment (sebaslogen) [inline core/navigation/src/main/kotlin/nl/q42/template/navigation/viewmodel/InitNavigator.kt:17]: Less business logic in the UI, more better 👍

File: feature/home/src/main/kotlin/nl/q42/template/presentation/home/second/HomeSecondViewModel.kt [hunk 2]
```diff
@@ -17,7 +18,7 @@ class HomeSecondViewModel @Inject constructor(
     private val titleParam = savedStateHandle.get<String>("title") ?: throw IllegalArgumentException("title required")
 
     private val _uiState = MutableStateFlow<HomeSecondViewState>(HomeSecondViewState(titleParam))
-    val uiState: Flow<HomeSecondViewState> = _uiState
+    val uiState: StateFlow<HomeSecondViewState> = _uiState
```
Comment (sebaslogen) [inline feature/home/src/main/kotlin/nl/q42/template/presentation/home/second/HomeSecondViewModel.kt:21]: Would you like to add .asStateFlow()?

**Review by sebaslogen — DISMISSED** [review]
Nice improvement, it's looking good and I have an improvement suggestion

File: feature/home/src/main/kotlin/nl/q42/template/presentation/home/HomeViewModel.kt [hunk 3]
```diff
@@ -21,7 +21,7 @@ class HomeViewModel @Inject constructor(
 ) : ViewModel(), RouteNavigator by navigator {
 
     private val _uiState = MutableStateFlow<HomeViewState>(HomeViewState.Empty)
-    val uiState: Flow<HomeViewState> = _uiState
+    val uiState: StateFlow<HomeViewState> = _uiState
```
[the remark this answers is not in the document]
Comment (ninovanhooff) [reply feature/home/src/main/kotlin/nl/q42/template/presentation/home/HomeViewModel.kt:24]: TIL

**Review by ninovanhooff — APPROVED** [review]
Nice!

**Pull request merged**